UpgradedVehicles/VehicleUpgrader.cs

```diff
@@ -324,26 +614,47 @@ private void UpdateSpeedRating(float speedBoosterCount)
         {
             this.SpeedMultiplier = GetSpeedMultiplierBonus(speedBoosterCount);
 
-            if (this.IsSeamoth)
+            if (ParentVehicle is Vehicle v)
             {
-                ParentVehicle.forwardForce = this.SpeedMultiplier * BaseSeamothSpeed;
-                ErrorMessage.AddMessage($"Seamoth Speed: {ParentVehicle.forwardForce}m/s ({this.SpeedMultiplier * 100f:00}%)");
+                if (v is SeaMoth)
+                {
+                    v.forwardForce = this.SpeedMultiplier * BaseSeamothSpeed;
+                    ErrorMessage.AddMessage($"Seamoth Speed: {v.forwardForce}m/s ({this.SpeedMultiplier * 100f:00}%)");
+                }
+                else if (v is Exosuit)
+                {
+                    v.forwardForce = this.SpeedMultiplier * BaseExosuitSwimSpeed;
+                    ErrorMessage.AddMessage($"Prawn Suit Water Speed: {v.forwardForce}m/s ({this.SpeedMultiplier * 100f:00}%)");
+                    v.onGroundForceMultiplier = this.SpeedMultiplier * BaseExosuitWalkSpeed;
+                    ErrorMessage.AddMessage($"Prawn Suit Land Speed: {v.onGroundForceMultiplier}m/s ({this.SpeedMultiplier * 100f:00}%)");
+
+                }
```

Comment:
hrmmm isnt the hoverbike a vehicle?? maybe if we are adding BZ support we should so something for it here?

Reply:
also I think sn1 has custom vehicles now that might benefit from some cross mod support?